x-pack/platform/plugins/shared/streams/server/lib/streams/state_management/streams/wired_stream.ts

```diff
@@ -440,8 +440,10 @@ export class WiredStream extends StreamActiveRecord<Streams.WiredStream.Definiti
     }
   }
 
-  protected async doDetermineCreateActions(): Promise<ElasticsearchAction[]> {
-    const ancestors = await this.dependencies.streamsClient.getAncestors(this._definition.name);
+  protected async doDetermineCreateActions(desiredState: State): Promise<ElasticsearchAction[]> {
+    const ancestors = getAncestorsAndSelf(this._definition.name).map(
+      (id) => desiredState.get(id)!.definition as Streams.WiredStream.Definition
```

Comment:
@flash1293 @miltonhultgren is there a case where an ancestor would not exist in the desiredState ? If a stream is updated or created, we have the guarantee that all its parent exist as they would either be already persisted or be created as a cascading change correct ?

Reply:
Huh, it should, but it doesn't - looks like we are only checking children downwards, we don't validate the parent is there.

Good catch. Since you are working on this right now, could you add a check for that to https://github.com/elastic/kibana/blob/de0440478b0e4e841b524bf97289fa3a47912d50/x-pack/platform/plugins/shared/streams/server/lib/streams/state_management/streams/wired_stream.ts#L274

Basically checking whether the parent of a wired stream existing in the desired state should be enough, since the routing stuff and so on is validated in the other direction already.

Reply:
Yeah we have a few statement where we defensively retrieve the parent from the desiredState ([example](https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/streams/server/lib/streams/state_management/streams/wired_stream.ts#L538)) and wanted to confirm that if the parent is not there it's actually a bug. So I'll remove these checks if that's the case